### PR TITLE
Fix reframe can handle single HTMLElement as a target

### LINKED ISSUE #149: Question: Is it intended that `target` can only be arrays?
## Examples
So we used to have `reframe(document.querySelector("iframe"))` and this worked fine, but since the new update this does not work. We're getting the following error:
```
reframe.es.js:26 Uncaught RangeError: Invalid array length
    at __spreadArrays (reframe.es.js:26)
    at reframe (reframe.es.js:42)
    ...
```

If I look at the error line I see the reframe `target` is no longer converted to an array. Instead it assumes the `target` is already an array.

This means we can change our code to:
- `reframe([document.querySelector("iframe")])` 
- `reframe(document.querySelectorAll("iframe"))`
- `reframe("iframe")`

**But is this expected behaviour, or do still want to allow the target to be a single object as well?**

### PR DESCRIPTION
**Pull request by GlennGeelen**
fixes #149

Change the way 'target' is handled in the reframe function.

This change is only made to the `reframe.ts`, I think the `noframe.ts` has the same problem, but not sure since we don't use that.

### CODE REVIEW

**Comment by GlennGeelen** [comment]
@yowainwright So I'm new to typescript so if you'd have any suggestions on how to improve the code, please suggest them.

File: src/reframe.ts [hunk 1]
```diff
@@ -5,14 +5,24 @@
  * @param cName
  * @summary defines the height/width ratio of the targeted <element>
  */
-export default function reframe(target: string | HTMLElement, cName: string = 'js-reframe') {
-  const frames = [...((typeof target === 'string' ? document.querySelectorAll(target) : target) as any)]
+export default function reframe(target: string | HTMLElement | NodeList, cName: string = 'js-reframe') {
+  let frames
```
Comment (yowainwright) [inline src/reframe.ts:9]:
```typescript
// prefer no let 😄 
const frames = typeof target === 'string' ? [...(document.querySelectorAll(target) as any)] : 'length' in target ? [...(target as any)] : [target];
```

File: src/reframe.ts [hunk 2]
```diff
@@ -5,14 +5,24 @@
...
+
+  if (typeof target === 'string') {
+    frames = [...(document.querySelectorAll(target) as any)]
+  } else if ('length' in target) {
+    frames = [...(target as any)]
+  } else {
+    frames = [target]
+  }
+
+  frames.forEach((frame) => {
```
Comment (yowainwright) [inline src/reframe.ts:20]: What does the compilation look like if we do `Array.from((frame) => {...}`? If it's the same, I'd prefer that. Otherwise, okay.

**Review by yowainwright — COMMENTED** [review]
Great stuff @GlennGeelen ! Just 2 comments... which your opinions are appreciated for. 

Thanks for your PR!!! :pray:

I'll update the TS to not use any after. You can see it after I do that. 😄

File: src/reframe.ts (hunk 2 above)
Comment (GlennGeelen) [reply src/reframe.ts:20]: I'll check

Comment (GlennGeelen) [reply src/reframe.ts:20]: Looks like `Array.from((frame) => {...}` is working 😄

Comment (GlennGeelen) [reply src/reframe.ts:20]: Looks like `from` is not working
```
(!) Plugin typescript: @rollup/plugin-typescript TS2339: Property 'from' does not exist on type 'any[]'.
```

**Comment by GlennGeelen** [comment]
> I'll update the TS to not use any after. You can see it after I do that. 

What do you mean exactly? 

Also, I'm not sure how to fix the codeFactor issues.

File: src/reframe.ts (hunk 2 above)
Comment (yowainwright) [reply src/reframe.ts:12]: By-pass my comment 😄  I'll approve with the update above.

**Comment by yowainwright** [comment]
> > I'll update the TS to not use any after. You can see it after I do that.
> 
> What do you mean exactly?
> > I'll update the types and CC you in the PR. TS can get tricky with iterating NodeLists 👍 
> 
> Also, I'm not sure how to fix the codeFactor issues.

For now, pass the target in as `any` and that should _maybe_ fix the codefactor issue. I will then go in and update type types.

**Comment by GlennGeelen** [comment]
@yowainwright I think I fixed all your comments 😄

**Comment by yowainwright** [comment]
@GlennGeelen Great work! I'll update types and publish this within the next day or so! Thank you so much!!!

**Pull request merged**